hakobera/embulk-input-mongodb
# Build with the "org.embulk.embulk-plugins" Gradle plugin

Hi @hakobera -san,

Thanks for contributing with the Embulk plugin over the open-source community!

In Embulk, we're recently recommending to use the `org.embulk.embulk-plugins` Gradle plugin so that:
* It can work with the latest Gradle easily
* It does not have to depend on the JRuby/Gradle plugin, which is not very stable
* It can be released as a valid Maven-based plugin distribution.
* It can manage its dependencies easily with the "Dependency lock" feature in Gradle.

See https://github.com/embulk/gradle-embulk-plugins for more information.

Along with that,
* Upgrade the Gradle wrapper to use Gradle 6.2.2
* Upgrade the Bintray Gradle plugin to 1.8.4 (latest)
* Upgrade depending embulk-core to 0.9.23 (v0.9 latest)
* Set JavaCompile option for deprecation and unchecked
* Catch up with the latest Embulk API specs and behaviors
  * Replace `Exec.getLogger` with `LoggerFactory.getLogger`
  * Replace Guava Optional to `java.util.Optional`
  * Remove use of Guava `Throwables.propagate`
  * Remove use of `@ConfigInject` by replacing with `Exec.getBufferAllocator`
  * Refactor the variable name with CommitReport to TaskReport

They are following Embulk's core-side changes described in: https://www.embulk.org/articles/2020/07/01/meetup-20200709.html

(This PR follows #54.)

Can you have a look when you have time?
